extension/InspectionDoc/src/Infrastructure/Reader/InspectionDocReader.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace InspectionDoc\Infrastructure\Reader;
+
+use InspectionDoc\Entity\InspectionDoc;
+use InspectionDoc\Entity\InspectionDocInterface;
+use InspectionDoc\Infrastructure\Loader\InspectionDocDataLoaderInterface;
+
+class InspectionDocReader implements InspectionDocReaderInterface
+{
+    /**
+     * @var array<string, \InspectionDoc\Entity\InspectionDocInterface>
+     */
+    protected array $inspectionDocsCache = [];
```

Comment:
Still not an option. I don't understand why you don't want to write testable code without any solid violations.   So far it's not a big deal, but it can cause some issues in the future. It is technical dept that we can fix now in a few minutes or later in a few hours or days.
But we already spent quite some time discussing this topic. Up to you.

Reply:
it's only for performance. There's no any logic and it should be incapsulated in current reader.